be/src/exec/topn_node.cpp

```diff
@@ -272,56 +278,60 @@ pipeline::OpFactories TopNNode::decompose_to_pipeline(pipeline::PipelineBuilderC
     // define a runtime filter holder
     context->fragment_context()->runtime_filter_hub()->add_holder(_id);
 
-    std::any context_factory;
-    if (is_partition) {
-        context_factory = std::make_shared<LocalPartitionTopnContextFactory>(
-                _tnode.sort_node.partition_exprs, _sort_exec_exprs.lhs_ordering_expr_ctxs(), _is_asc_order,
-                _is_null_first, _sort_keys, _offset, partition_limit, _tnode.sort_node.topn_type, _order_by_types,
-                _materialized_tuple_desc, child(0)->row_desc(), _row_descriptor);
-    } else {
-        context_factory = std::make_shared<SortContextFactory>(
-                runtime_state(), _tnode.sort_node.topn_type, is_merging, _offset, _limit,
-                _sort_exec_exprs.lhs_ordering_expr_ctxs(), _is_asc_order, _is_null_first, _build_runtime_filters);
+    auto degree_of_parallelism = context->source_operator(ops_sink_with_sort)->degree_of_parallelism();
+
+    // spill components
+    // TODO: avoid create spill channel when when disable spill
+
+    auto executor = std::make_shared<IOTaskExecutor>(ExecEnv::GetInstance()->pipeline_sink_io_pool());
+    auto spill_channel_factory =
+            std::make_shared<SpillProcessChannelFactory>(degree_of_parallelism, std::move(executor));
+
+    // spill process operator
+    if (runtime_state()->enable_spill() && _limit < 0) {
```

Comment:
limit < 0?

Reply:
now we only support spill for full-sort. so _limit should be -1